pkg/intent/intent.go

```diff
@@ -126,3 +127,16 @@ func (i *Store) SetPod(node string, manifest pods.PodManifest) (time.Duration, e
 	}
 	return writeMeta.RequestTime, err
 }
+
+func (i *Store) RegisterPodService(manifest pods.PodManifest) error {
+	podService := &consulapi.AgentServiceRegistration{
```

Comment:
It should be. If you register two services with the same ID on the same node, consul overwrites the older one with the newer one (which is desirable behavior for us, I think). Unfortunately this behavior is not documented, but I found it somewhere in their codebase. Let me find a link.

Repeatedly registering a service over and over might have an effect on the timing of its health checks (ie maybe the health check gets fired every time you register) but I don't think that's a significant concern.